release-22.2: server: stop grouping stmts by transaction_fingerprint_id

xinhaoz opened the pull request:
Backport 4/4 commits from #99719.

/cc @cockroachdb/release

---

See individual commits.

Fixes: #99708
Fixes: #99390

DB-Console: https://www.loom.com/share/d1f19c1623e7439f8ace836b2cbb2494


Release justification: bug fix

xinhaoz committed:
ui: stop aggregating fingerprints response on UI

In the combined stats EP, we group entries on both fingerprint_id
and app_name for the overview page queries. The client has been
grouping only on fingerprint_id for both overview pages. With the
introduction of the limit param, this mismatch in aggregation criteria
can lead to fewer than the requested rows returned from the server.

This commit ensures that the UI does no additional aggregation with
the payload from the server. In the next commit, we will provide the
app name of the selected row when routing to statement fingerprint
details from the overview page, so that we only req stats for the
app of the selected row.

As a part of this change, we create a common combiner body for the stmts
selector in both db-console and cluster-ui to reduce duplicated code.
Relevant aggregation code on the client has also been removed.

Part of: #99708
Fixes: #99390

Release note (ui change): By default, we now show the `Application Name`
column in the fingerprints overview pages. Stmts and txns fingerprints
will be displayed per application on the overview pages rather than
grouped into a single fingerprint id.

xinhaoz committed:
server: stop grouping stmts by transaction_fingerprint_id

Previously, the query for the stmt fingerprints overview page
grouped stmts on transaction_fingerprint_id. We should remove
this grouping criteria since we do not care about the txn at
this page level, and including it means we can't return as many
unique stmt fingerprint ids on the page. To view txn specific
stats / stmts for a txn, users can use the txn fingerprints page.
We add the distinct txn_fingerprint_ids for the grouped stmts as
a new field for testing purposes. Note that the
`transaction_fingerprint_id` in the key_data field for stmts returned
by the CombinedStatements api will now be blank.

New proto fields:
- txn_fingerprint_ids (for StatementsResponse_CollectedStatementStatistics)

Part of: #99708

Release note: None

xinhaoz committed:
ui: use app name of selected row when fetching fingerprint details

Previously, the fingerprint details pages would show the details
aggregated by the selected app names (from the filter, or aggregated
across all apps if none were selected). Since we no longer perform
any aggregation in the overview pages we now always display each
fingerprint attached to a single app for each row. When going to
the details page via clicking a row from the overview page, we
should fetch details only for the selected app.

Note that the txn fingerprint details page simply looks for a
corresponding entry from the overview response, thus we change
that search criteria to incude the app name of the selected row.

The app name has also been included in  the link to stmt details
from insights details.

Release note (ui change): When going from the fingerprint overview
pages or the insight details pages to the fingerprint details
page, the details page will fetch data for the stmt with the
provided application name. For overview pages, this is the app
name of the selected row. For insight details, it is the app of
the execution that generated the insight.

Part of: #99708

xinhaoz committed:
ui: remove `totalFingerprints` prop from statement page

This commit removes the unused prop `totalFingerprints` and its
relevant selectors.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/101475)
<!-- Reviewable:end -->

xinhaoz committed:
ui: add aggregation step for cases when we have an older server version

For CC clusters we may be in a situation where we have the latest
cluster-ui version for a major version, but an older server version.
In those cases, the stmt stats data returned by the server will be
grouped on more than just the stmt fingerprint id and app name, so
we should put an additional grouping step in the UI payload handler
to ensure that the data is aligned with what is expected by the
newer UI version.

Epic: none

Release note: None

maryliag reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 4 files at r1, 13 of 13 files at r2, 12 of 12 files at r3, 6 of 6 files at r4, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/101475#-:-NTFsC_W5D4JACLzNPU3:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @xinhaoz)



<!-- Sent from Reviewable.io -->
